### PR TITLE
Update 10-route-handlers.mdx

### PR DESCRIPTION
**Pull request by pkellner**
TypeScript creates an error if you don't explicitly set a header value. undefined is no good.

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 18bb37e884d704f7435cc8d73540ffc2df748f45

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by pkellner** [comment]
> Allow CI Workflow Run
> 
> * [ ]  approve CI run for commit: [18bb37e](https://github.com/vercel/next.js/commit/18bb37e884d704f7435cc8d73540ffc2df748f45)
> 
> Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

I'm not following @ijjk, is this really you commenting or a bot somehow. If really you, what is "this" in "Note: `this` should only be enabled once the ..."

File: docs/02-app/01-building-your-application/01-routing/10-route-handlers.mdx [hunk 1]
```diff
@@ -132,7 +132,7 @@ export async function POST() {
     method: 'POST',
     headers: {
       'Content-Type': 'application/json',
-      'API-Key': process.env.DATA_API_KEY,
+      'API-Key': process.env.DATA_API_KEY ?? 'not-found',
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/10-route-handlers.mdx:135]:
```suggestion
      'API-Key': process.env.DATA_API_KEY!,
```

File: docs/02-app/01-building-your-application/01-routing/10-route-handlers.mdx [hunk 2]
```diff
@@ -99,7 +99,7 @@ export async function GET(request: Request) {
   const res = await fetch(`https://data.mongodb-api.com/product/${id}`, {
     headers: {
       'Content-Type': 'application/json',
-      'API-Key': process.env.DATA_API_KEY,
+      'API-Key': process.env.DATA_API_KEY ?? 'not-found',
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/10-route-handlers.mdx:102]:
```suggestion
      'API-Key': process.env.DATA_API_KEY!,
```

**Review by leerob — APPROVED** [review]

**Pull request merged**